content/en/functions/templates/Current.md

```diff
@@ -0,0 +1,127 @@
+
+Filename
+: (`string`) Returns the absolute path of the current template. This will be empty for embedded templates.
+
+Name
+: (`string`) Returns the path of the current template, relative to the `layouts` directory.
+
+Parent
+: (`tpl.CurrentTemplateInfo`) Returns an object representing the parent of the current template, if any. This may be `nil`.
+
+## Examples
+
+The examples below help visualize template execution and require a `debug` parameter set to `true` in your site configuration:
+
+{{< code-toggle file=hugo >}}
+[params]
+debug = true
+{{< /code-toggle >}}
+
+### Template boundaries
+
+To visually mark where a template begins and ends execution:
+
+```go-html-template {file="layouts/_default/single.html"}
+{{ define "main" }}
+  {{ if site.Params.debug }}
+    <div class="debug">[entering {{ templates.Current.Filename }}]</div>
+  {{ end }}
+
+  <h1>{{ .Title }}</h1>
+  {{ .Content }}
+
+  {{ if site.Params.debug }}
+    <div class="debug">[leaving {{ templates.Current.Filename }}]</div>
+  {{ end }}
+{{ end }}
+```
+
+### Template call stack
+
+To show the chain of templates that led to the current one, create a partial template that iterates through its ancestors:
+
+```go-html-template {file="layouts/partials/render-template-stack.html" copy=true}
+<div class="debug">
+  {{ with templates.Current }}
+    {{ range .Ancestors }}
+      {{ .Filename }}<br>
+      {{ with .Base }}
+        {{ .Filename }}<br>
+      {{ end }}
+    {{ end }}
+  {{ end }}
+</div>
+```
+
+Then call the partial from any template:
+
+```go-html-template {file="layouts/partials/footer/copyright.html" copy=true}
+{{ if site.Params.debug }}
+  {{ partial "render-template-stack.html" . }}
+{{ end }}
+```
+
+The rendered template stack would look something like this:
+
+```text
+/home/user/project/layouts/partials/footer/copyright.html
+/home/user/project/themes/foo/layouts/partials/footer.html
+/home/user/project/themes/foo/layouts/_default/single.html
+/home/user/project/themes/foo/layouts/_default/baseof.html
+```
+
+To reverse the order of the entries chain the `Reverse` method to the `Ancestors` method:
+
+```go-html-template {file="layouts/partials/render-template-stack.html" copy=true}
+<div class="debug">
+  {{ with templates.Current }}
+    {{ range .Ancestors.Reverse }}
+      {{ with .Base }}
+        {{ .Filename }}<br>
+      {{ end }}
+      {{ .Filename }}<br>
+    {{ end }}
+  {{ end }}
+</div>
+```
+
+To render each entry as an anchor element that, when clicked, will open the template in VS Code:
+
+```go-html-template {file="layouts/partials/render-template-stack.html" copy=true}
+<div class="debug">
+  {{ with templates.Current }}
+    {{ range .Ancestors }}
+      <a href="vscode://file/{{ .Filename }}">{{ .Filename }}</a><br>
+      {{ with .Base }}
+        <a href="vscode://file/{{ .Filename }}">{{ .Filename }}</a><br>
+      {{ end }}
+    {{ end }}
+  {{ end }}
+</div>
```

Comment:
These are all good examples. One I suspect would be useful is something ala this `current.html` partial:

```handlebars
{{ with templates.Current.Parent }}
  <div class="flex">
    <a
      href="vscode://file/{{ .Filename }}"
      class="text-blue-600 hover:text-blue-500"
      >{{ .Name }}</a
    >
    {{ with .Base }}
      <a
        href="vscode://file/{{ .Filename }}"
        class="text-blue-600 hover:text-blue-500 ml-4"
        >{{ .Name }}</a
      >
    {{ end }}
  </div>
{{ end }}
```